src/lib/QueryType.hs

```diff
@@ -615,6 +625,11 @@ getTypePrimOp op = case op of
   OutputStreamPtr ->
     return $ BaseTy $ hostPtrTy $ hostPtrTy $ Scalar Word8Type
     where hostPtrTy ty = PtrType (Heap CPU, ty)
+  Perform hndDict i -> do
```

Comment:
Huh, I thought that `Perform` should also take an argument and actually perform the effect. But it seems that it really acts more like a projection of the method impl from the handler, right? In that case, maybe rename it to `GetEffOp` or something else along those lines